tools/ci/run_tc.py

```diff
@@ -0,0 +1,122 @@
+#!/usr/bin/env python
+
+import argparse
+import json
+import os
+import re
+import subprocess
+import sys
+
+
+root = os.path.abspath(
+    os.path.join(os.path.dirname(__file__),
+                 os.pardir,
+                 os.pardir))
+
+
+def run(cmd, return_stdout=False, **kwargs):
+    print(" ".join(cmd))
+    if return_stdout:
+        f = subprocess.check_output
+    else:
+        f = subprocess.check_call
+    return f(cmd, **kwargs)
+
+
+def get_parser():
+    p = argparse.ArgumentParser()
+    p.add_argument("job",
+                   help="Name of the job associated with the current event")
+    p.add_argument("script", help="Script to run for the job")
+    return p
+
+
+def get_extra_jobs(event):
+    body = None
+    jobs = set()
+    if "commits" in event:
+        body = event["commits"][0]["message"]
+    elif "pull_request" in event:
```

Comment:
Can we not make this an if?

Can we make it support both commit message and PR body? I think it ignoring commit message is surprising in the PR case?

Reply:
It typically works OK anyway because the default PR body is the commit message when there's only one push. In any case just making this an if doesn't work because the PR event doesn't contain commits so we'd have to actually check with git, which I would at least like to avoid at first.

Reply:
Ah, OK. :/